extensions.yaml

```diff
@@ -31,6 +31,9 @@ extensions:
   pypi: datalad-webapp
   package: datalad_webapp
   tester: pytest
+ datalad_osf:
+  pip_depends: [pytest]
```

Comment:
why `pytest` dependency but then 
```
python -m nose -s -v --with-cov --cover-package datalad datalad_osf.tests
```
should `tester: pytest` be added if you use pytest?

Also note that the workflow is a "generated" file -- it must not be edited manually.

Reply:
> Also note that the workflow is a "generated" file -- it must not be edited manually.

How are secrets supposed to be added? They need to go into the workflow file as far as I know?
https://docs.github.com/en/free-pro-team@latest/actions/reference/encrypted-secrets

Reply:
GOOD POINT!  
- Ideally we should allow in the spec to list secrets to expose and provision for that in the template. So everything alike for any other optionally included part of the workflow, e.g. for apt_depends: https://github.com/datalad/datalad-extensions/blob/master/templates/.github/workflows/test-%7B%7Bextension.name%7D%7D.yaml#L69
- meanwhile, if not feeling up to implementing above (but I really hope that you don't succumb to this ;)), I guess there should be no harm if we expose secrets even within unrelated workflows by just tuning the main template to always include them https://github.com/datalad/datalad-extensions/blob/master/templates/.github/workflows/test-%7B%7Bextension.name%7D%7D.yaml#L76